stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanActivity.kt

```diff
@@ -0,0 +1,683 @@
+     */
+    private fun prepareCamera(onCameraReady: () -> Unit) {
+        viewBinding.previewFrame.post {
+            viewBinding.viewFinderBackground
+                .setViewFinderRect(viewBinding.viewFinderWindow.asRect())
+            onCameraReady()
+        }
+    }
+
+    private fun onCameraReady() {
+        cameraAdapter.bindToLifecycle(this)
+
+        val torchStat = Stats.trackTask("torch_supported")
+        cameraAdapter.withFlashSupport {
+            launch { torchStat.trackResult(if (it) "supported" else "unsupported") }
+            setFlashlightState(cameraAdapter.isTorchOn())
+            viewBinding.torchButton.setVisible(it)
+        }
+
+        cameraAdapter.withSupportsMultipleCameras {
+            viewBinding.swapCameraButton.setVisible(it)
+        }
+
+        launch { onCameraStreamAvailable(cameraAdapter.getImageStream()) }
+    }
+
+    /**
+     * Turn the flashlight on or off.
+     */
+    private fun setFlashlightState(on: Boolean) {
+        cameraAdapter.setTorchState(on)
+        isFlashlightOn = on
+        if (isFlashlightOn) {
+            viewBinding.torchButton.setDrawable(R.drawable.stripe_flash_on_dark)
+        } else {
+            viewBinding.torchButton.setDrawable(R.drawable.stripe_flash_off_dark)
+        }
+    }
+
+    /**
+     * Set the focus of the camera.
+     */
+    private fun setFocus(point: PointF) {
+        cameraAdapter.setFocus(point)
+    }
+
+    /**
+     * Generate a camera adapter
+     */
+    private fun buildCameraAdapter(): CameraAdapter<CameraPreviewImage<Bitmap>> =
+        getCameraAdapter(
+            activity = this,
+            previewView = viewBinding.previewFrame,
+            minimumResolution = MINIMUM_RESOLUTION,
+            cameraErrorListener = cameraErrorListener,
+        )
+
+    private fun ensureValidParams() = when {
+        params.stripePublishableKey.isEmpty() -> {
+            scanFailure(InvalidStripePublishableKeyException("Missing publishable key"))
+            false
+        }
+//        params.cardImageVerificationIntentId.isEmpty() -> {
+//            scanFailure(InvalidCivException("Missing card image verification ID"))
+//            false
+//        }
+//        params.cardImageVerificationIntentSecret.isEmpty() -> {
+//            scanFailure(InvalidCivException("Missing card image verification client secret"))
+//            false
+//        }
+        else -> true
+    }
+
+    private fun closeScanner() {
+//        uploadScanStats(
+//            stripePublishableKey = params.stripePublishableKey,
+//            civId = params.cardImageVerificationIntentId,
+//            civSecret = params.cardImageVerificationIntentSecret,
+//            instanceId = Stats.instanceId,
+//            scanId = Stats.scanId,
+//            device = Device.fromContext(this),
+//            appDetails = AppDetails.fromContext(this),
+//            scanStatistics = ScanStatistics.fromStats()
+//        )
```

Comment:
Let me check with the rest of the team to see if we can omit scan stats. If we can, we won't need the `stripePublishableKey` any more. If we can't, we should uncomment this block.

See https://stripe.slack.com/archives/C022248ERA9/p1643226050053700 for discussion on this.